tools/content/fetch_messages.dart

```diff
@@ -0,0 +1,223 @@
+///
+/// This script is meant to be run via `tools/content/check-features`.
+///
+/// For more help, run `tools/content/fetch_message.dart --help`.
+///
+/// See also:
+/// * tools/content/unimplemented_features_test.dart, which runs checks against
+///   the fetched corpuses.
+void main(List<String> args) async {
+  final argParser = ArgParser();
+  argParser.addOption(
+    'config-file',
+    help: 'A zuliprc file with identity information including email, API key\n'
+          'and the Zulip server URL to fetch the messages from (required).\n\n'
+          'To get the file, see\n'
+          'https://zulip.com/api/configuring-python-bindings#download-a-zuliprc-file.',
+    valueHelp: 'path/to/zuliprc',
+  );
+  argParser.addOption(
+    'corpus-dir',
+    help: 'The directory to look for/store the corpus file.  If not given,\n'
+          'the script will write output to stdout.  Otherwise, this will\n'
+          'first read from the existing corpus file (assumed to be named\n'
+          'as "host name of the Zulip server.jsonl") to avoid duplicates\n'
+          'before fetching more messages',
+    valueHelp: 'path/to/czo.jsonl',
+  );
+  argParser.addFlag(
+    'fetch-newer',
+    help: 'Fetch newer messages instead of older ones.\n'
+          'Only useful when there is a matching corpus file in corpus-dir.',
+    defaultsTo: false,
+  );
+  argParser.addFlag(
+    'help', abbr: 'h',
+    negatable: false,
+    help: 'Show this help message.',
+  );
+
+  void printUsage() {
+    // Give it a pass when printing the help message.
+    // ignore: avoid_print
+    print('usage: fetch_messages --config-file <CONFIG_FILE>\n\n'
+          'Fetch message contents from a Zulip server in bulk.\n\n'
+          '${argParser.usage}');
+  }
+
+  Never throwWithUsage(String error) {
+    printUsage();
+    throw Exception('\nError: $error');
+  }
+
+  final parsedArguments = argParser.parse(args);
+  if (parsedArguments['help'] as bool) {
+    printUsage();
+    exit(0);
+  }
+
+  final zuliprc = parsedArguments['config-file'] as String?;
+  if (zuliprc == null) {
+    throwWithUsage('"config-file is required');
+  }
+
+  final configFile = File(zuliprc);
+  if (!configFile.existsSync()) {
+    throwWithUsage('Config file "$zuliprc" does not exist');
+  }
+
+  // `zuliprc` is a file in INI format containing the user's identity
+  // information.
+  //
+  // See also:
+  //   https://zulip.com/api/configuring-python-bindings#configuration-keys-and-environment-variables
+  final parsedConfig = ini.Config.fromString(configFile.readAsStringSync());
+  final email = parsedConfig.get('api', 'email') as String;
+  final apiKey = parsedConfig.get('api', 'key') as String;
+  final site = Uri.parse(parsedConfig.get('api', 'site') as String);
+
+  final outputDirStr = parsedArguments['corpus-dir'] as String?;
+  final fetchNewer = parsedArguments['fetch-newer'] as bool;
+  int? anchorMessageId;
```

Comment:
In a CLI program I think it's generally helpful to split the parsing of the command line (and of config that's closely tied to the command line) from the rest of the logic.

So between these two lines is basically where that transition happens — we're done inspecting `parsedArguments` (and `parsedConfig`), and we're starting to do the work and to have side effects. It'd therefore be good to take the bottom half of the function, below this point, and move it to its own function; `main` would end by calling that function, passing a bunch of arguments like `email` and `outputDirStr`.